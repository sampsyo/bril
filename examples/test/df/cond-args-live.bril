# ARGS: live

@main(cond: bool) {
  a: int = const 47;
  b: int = const 42;
  br cond .left .right;
.left:
  b: int = const 1;
  c: int = const 5;
  jmp .end;
.right:
  a: int = const 2;
  c: int = const 10;
  jmp .end;
.end:
  d: int = sub a c;
  print d;
}
