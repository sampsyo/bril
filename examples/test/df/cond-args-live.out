b1:
  in:  cond
  out: a
left:
  in:  a
  out: a, c
right:
  in:  ∅
  out: a, c
end:
  in:  a, c
  out: ∅
